docs/internals/optimizer.rst

```diff
@@ -1096,6 +1097,48 @@ The step LiteralRematerialiser is not required for correctness. It helps deal wi
 ``function f(x) -> y { revert(y, y} }`` where the literal ``y`` will be replaced by its value ``0``,
 allowing us to rewrite the function.
 
+.. _unused-store-eliminator:
+
+UnusedStoreEliminator
+^^^^^^^^^^^^^^^^^^^^^
+
+Optimizer component that removes ``sstore`` and ``mstore`` statements if conditions are met for said removal.
+In case of an ``sstore``, if all outgoing code paths lead to a revert or to another ``sstore`` for which the optimizer
+can tell that it will overwrite the first store, the statement will be removed. For example, the following code
+
+.. code-block:: yul
+
+    {
+        let c := calldataload(0)
+        sstore(c, 1)
+        if c {
+            sstore(c, 2)
+        }
+        sstore(c, 3)
+    }
```

Comment:
Another `sstore` that writes to `c`, or to a different variable?

Reply:
On the other hand, I think the examples should be as simple (minimal) as possible, so not sure what adding another non-redundant `sstore` brings, since the last `sstore(c, 3)` is already not redundant and is hence not removed after the step?